spec/system/assign_topic_spec.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+describe "Assign | Assigning topics", type: :system, js: true do
+  let(:topic_page) { PageObjects::Pages::Topic.new }
+  let(:assign_modal) { PageObjects::Modals::Assign.new }
+  fab!(:staff_user) { Fabricate(:user, groups: [Group[:staff]]) }
+  fab!(:admin) { Fabricate(:admin) }
+  fab!(:topic) { Fabricate(:topic) }
+  fab!(:post) { Fabricate(:post, topic: topic) }
+
+  before do
+    SiteSetting.assign_enabled = true
+    sign_in(admin)
+  end
+
+  describe "with open topic" do
+    it "can assign and unassign" do
+      visit "/t/#{topic.id}"
+
+      topic_page.click_assign_topic
+      assign_modal.set_assignee(staff_user)
+      assign_modal.confirm
+
+      expect(topic_page).to have_assignment_action(2, "assigned", staff_user)
+      expect(find("#topic .assigned-to")).to have_content(staff_user.username)
+
+      topic_page.click_unassign_topic
+
+      expect(topic_page).to have_assignment_action(3, "unassigned", staff_user)
+      expect(page).to have_no_css("#topic .assigned-to")
+    end
+
+    context "when unassign_on_close is set to true" do
+      before { SiteSetting.unassign_on_close = true }
+
+      it "topic gets unassigned on close" do
+        visit "/t/#{topic.id}"
+
+        topic_page.click_assign_topic
+        assign_modal.set_assignee(staff_user)
+        assign_modal.confirm
+
+        expect(topic_page).to have_assignment_action(2, "assigned", staff_user)
+
+        find(".topic-footer-main-buttons .toggle-admin-menu").click
+        find(".topic-admin-close").click
+
+        expect(find("#post_3")).to have_content(
+          I18n.t("js.action_codes.closed.enabled", when: "just now"),
+        )
+        expect(page).to have_no_css("#post_4")
+        expect(page).to have_no_css("#topic .assigned-to")
+      end
+
+      it "can assign the previous assignee" do
+        visit "/t/#{topic.id}"
+
+        topic_page.click_assign_topic
+        assign_modal.set_assignee(staff_user)
+        assign_modal.confirm
+
+        expect(topic_page).to have_assignment_action(2, "assigned", staff_user)
+
+        find(".topic-footer-main-buttons .toggle-admin-menu").click
+        find(".topic-admin-close").click
+
+        expect(find("#post_3")).to have_content(
+          I18n.t("js.action_codes.closed.enabled", when: "just now"),
+        )
+        expect(page).to have_no_css("#post_4")
+        expect(page).to have_no_css("#topic .assigned-to")
+
+        topic_page.click_assign_topic
+        assign_modal.set_assignee(staff_user)
```

Comment:
Maybe we could use ruby at our advantage here and defined `def assignee=(user)` instead of `set_assignee`